lang/csharp/src/apache/main/File/Codec.cs

```diff
@@ -43,9 +44,10 @@ public abstract class Codec
         /// <summary>
         /// Decompress data using implemented codec
         /// </summary>
-        /// <param name="compressedData"></param>
+        /// <param name="compressedData">The buffer holding data to decompress.</param>
+        /// <param name="length">The actual length of bytes to decompress from the buffer.</param>
         /// <returns></returns>
-        abstract public byte[] Decompress(byte[] compressedData);
+        abstract public byte[] Decompress(byte[] compressedData, int length);
```

Comment:
Can we preserve the old method for backward compatibility ?
E.g. 
```C#
public byte[] Decompress(byte[] compressedData) {
   return Decompress(compressedData.length);
}
```

Reply:
Done! Thanks Martin

Reply:
Since you force pushed it is hard to see how you improved it.
Now we have two similar methods in the API and at least to me it is not very clear when to use each one.
I'll let someone else who has experience with C# to decide whether this PR should be merged.

Reply:
I thought I had read somewhere in the contribution guidelines that a single commit per issue was preferred. However, I've read the guidelines again and I cannot find it. Is the preference to have multiple commits and squash merge at the end?

The change I've pushed is simply to overload the method so that you have both method signatures, the one without a length argument and the one with it.

Reply:
The best is to use new commits after a review. Otherwise we have to re-review the whole thing again and again.
Once the PR is approved we use GitHub's "Squash & Merge" button. This way all the commit messages are preserved for the final commit. So, better use good commit messages! :-)

Reply:
> The change I've pushed is simply to overload the method so that you have both method signatures, the one without a length argument and the one with it.

I still think my suggestion to mark the old method as deprecated/Obsolete and to implement it as in https://github.com/apache/avro/pull/1358/files#r795248968 better.
Currently https://github.com/apache/avro/pull/1358/files#diff-f6b2b3f158e175d309773571f11f7cf979b6ba983e3c00f28b0d7e869910f6b3R310 calls the new method. So, at some point we should be able to remove the old one.
I don't know whether C# supports `default method in interface` as in Java/Scala. It would be the best if the old method is left as Obsolete only in Codec.cs

Reply:
Right, I assumed you were concerned about library consumers who could potentially still be using `Codec` as it's marked `public` so I avoided making changes that could break it in any way (including marking as obsolete as that can break builds if they treat warnings as errors).
I have pushed the changes now so that the old method is `Obsolete`. Thanks!